assets/js/plugin-check-admin.js

```diff
@@ -0,0 +1,49 @@
+( function ( data ) {
+	const checkItButton = document.getElementById( 'plugin-check__submit' );
+	const pluginsList   = document.getElementById( 'plugin-check__plugins-dropdown' );
+
+	// Return early if the elements cannot be found on the page.
+	if ( ! checkItButton || ! pluginsList ) {
+		console.error( 'Missing form elements on page' );
+		return;
+	}
+
+	checkItButton.addEventListener( 'click', (e) => {
+		e.preventDefault();
+
+		const pluginCheckData = new FormData();
+
+		// Collect the data to pass along for generating a check results.
+		pluginCheckData.append( 'action', 'plugin_check_run_checks' );
+		pluginCheckData.append( 'nonce', data.nonce );
+		pluginCheckData.append( 'plugin', pluginsList.value );
+
+		fetch(
+			data.ajaxUrl,
+			{
+				method: 'POST',
+				credentials: 'same-origin',
+				body: pluginCheckData
+			}
+		)
+		.then(
+			( response ) => {
+				return response.json();
+			}
+		)
+		.then(
+			( data ) => {
+				if ( ! data.success ) {
+					throw new Error( data.data[0].message );
+				}
```

Comment:
Have you tested that this works? What if no response is returned, e.g. when no network connection (thus `data` could be `null` or `undefined`)? Let's use defensive coding here to make triple-sure that `data.success` and `data.data[ 0 ].message` are set before accessing it. We should avoid errors in our error handling logic :)

Last but not least: Have you tested that this logic is actually reached when an error happens? Since the response code is a 403, will it get to the code here, or would it immediately run into the `catch` logic without even running this?

Reply:
Thanks @felixarntz I've updated and tested this based on your feedback. To answer some of your questions here:

- From what I understand, if there is no network connection the response is not handled and by the rest of the code and a generic console error is displayed.
- Recent updates check the `response.ok` property before moving on and will throw an error with the status code and text. This now includes the 403 error we return. Originally, the custom invalid nonce error message would be console logged.
- I have tested for different scenarios (500 error, 403 error, returning no data, successful response) and the code is working.

Reply:
Thanks @jjgrainger for confirming!